### PR TITLE
LR-186:Get rid of pending tests

### PR DESCRIPTION
**Pull request by superNazo**
[Link to JIRA ticket](https://ssu-jira.softserveinc.com/secure/RapidBoard.jspa?rapidView=484&view=detail&selectedIssue=LR-186)

### Author.
Nazarii Ivankiv

- [x] I have added link to this PR to the JIRA
- [x] I have logged time in JIRA.

### Reviewers.

- [x] @sergiy17 
- [x] @anatoliiibrahimov 
- [x] @vovka

### Issue summary.

There were a bunch of pending tests

### Fix summary. 

Tests work, there is no pendings, coverage is 98.2 percents

### CODE REVIEW

**Comment by anatoliiibrahimov** [comment]
nice :+1:

File: spec/controllers/quizzes_controller_spec.rb [hunk 1]
```diff
@@ -1,40 +1,20 @@
```
Comment (YUzhva) [inline spec/controllers/quizzes_controller_spec.rb]: empty line?)

File: spec/controllers/quizzes_controller_spec.rb [hunk 2]
```diff
@@ -47,15 +27,13 @@
     end
 
     it "redirect to home page if not sign in" do
+      sign_out @user
       get :index, {}, valid_session
```
Comment (YUzhva) [inline spec/controllers/quizzes_controller_spec.rb]: empty line?)

File: spec/controllers/quizzes_controller_spec.rb [hunk 3]
```diff
@@ -47,15 +27,13 @@
...
 
       expect(response.status).to eq 302
     end
 
     it "if user sign in show quizzes" do
-      user = User.create!(email: "user@example.org", name: "Name")
-      sign_in user
-
       get :index, {}, valid_session
```
Comment (YUzhva) [inline spec/controllers/quizzes_controller_spec.rb]: empty line?)

File: spec/controllers/quizzes_controller_spec.rb [hunk 4]
```diff
@@ -72,9 +50,6 @@
 
   describe "GET #new" do
     it "assigns a new quiz as @quiz" do
-      user = User.create!(email: "user@example.org", name: "Name")
-      sign_in user
-
       get :new
```
Comment (YUzhva) [inline spec/controllers/quizzes_controller_spec.rb]: empty line?)

File: spec/controllers/quizzes_controller_spec.rb [hunk 5]
```diff
@@ -125,14 +100,14 @@
...
-        skip("Add a hash of attributes valid for your model")
+        { name: 'update name' }
       }
 
       it "updates the requested quiz" do
         quiz = Quiz.create! valid_attributes
         put :update, {:id => quiz.to_param, :quiz => new_attributes}, valid_session
         quiz.reload
-        skip("Add assertions for updated state")
+        expect(response).to redirect_to(quiz)
```
Comment (vovka) [inline spec/controllers/quizzes_controller_spec.rb]: I don't think this assertion is correct. You assert redirect when checking for update - it's not consistent. How do you think what assertion could be better here?

File: spec/controllers/quizzes_controller_spec.rb [hunk 6]
```diff
@@ -1,40 +1,20 @@
...
-# sticking to rails and rspec-rails APIs to keep things simple and stable.
-#
-# Compared to earlier versions of this generator, there is very limited use of
-# stubs and message expectations in this spec.  Stubs are only used when there
-# is no simpler way to get a handle on the object needed for the example.
-# Message expectations are only used when there is no simpler way to specify
-# that an instance is receiving a specific message.
-
 RSpec.describe QuizzesController, type: :controller do
+  before(:each) {
```
Comment (vovka) [inline spec/controllers/quizzes_controller_spec.rb]: Prefer using `do...end` with multiline code blocks.

File: spec/controllers/quizzes_controller_spec.rb [hunk 7]
```diff
@@ -47,15 +27,13 @@
     end
 
     it "redirect to home page if not sign in" do
+      sign_out @user
```
Comment (vovka) [inline spec/controllers/quizzes_controller_spec.rb]: Coul you separate this setup phase with an empty line as well?

File: spec/controllers/quizzes_controller_spec.rb (hunk 2 above)
Comment (vovka) [inline spec/controllers/quizzes_controller_spec.rb]: It's a kind of test phases separation.

File: spec/controllers/quizzes_controller_spec.rb (hunk 1 above)
Comment (superNazo) [inline spec/controllers/quizzes_controller_spec.rb]: yes, I've missed this

File: spec/controllers/quizzes_controller_spec.rb (hunk 6 above)
Comment (superNazo) [inline spec/controllers/quizzes_controller_spec.rb]: ok

File: spec/controllers/quizzes_controller_spec.rb (hunk 7 above)
Comment (superNazo) [inline spec/controllers/quizzes_controller_spec.rb]: ok

File: spec/controllers/quizzes_controller_spec.rb (hunk 2 above)
Comment (superNazo) [inline spec/controllers/quizzes_controller_spec.rb]: I'll fix it

File: spec/controllers/quizzes_controller_spec.rb (hunk 5 above)
Comment (superNazo) [inline spec/controllers/quizzes_controller_spec.rb]: maybe this one 
expect(assigns(:quiz)).to eq(quiz)
?

File: spec/controllers/quizzes_controller_spec.rb [hunk 8]
```diff
@@ -125,39 +108,49 @@
...
       }
 
       it "updates the requested quiz" do
         quiz = Quiz.create! valid_attributes
+
         put :update, {:id => quiz.to_param, :quiz => new_attributes}, valid_session
         quiz.reload
-        skip("Add assertions for updated state")
+
+        expect(assigns(:quiz)).to eq(quiz)
```
Comment (vovka) [inline spec/controllers/quizzes_controller_spec.rb]: I still have doubts regarding this test.

**Comment by vovka** [comment]
Looks good to me

File: spec/controllers/quizzes_controller_spec.rb [hunk 9]
```diff
@@ -125,39 +108,48 @@
...
+        { name: 'updated name' }
       }
 
       it "updates the requested quiz" do
         quiz = Quiz.create! valid_attributes
+
         put :update, {:id => quiz.to_param, :quiz => new_attributes}, valid_session
-        quiz.reload
-        skip("Add assertions for updated state")
+
```
Comment (vovka) [inline spec/controllers/quizzes_controller_spec.rb]: Codacy Issue found: [`(...)` interpreted as grouped expression.](https://www.codacy.com/app/vovka/video_quiz/file/2443476934/issues/source?bid=3205144&fileBranchId=3243807#l119)

Comment (vovka) [inline spec/controllers/quizzes_controller_spec.rb]: Codacy Issue found: [`(...)` interpreted as grouped expression.](https://www.codacy.com/app/vovka/video_quiz/file/2443476934/issues/source?bid=3205144&fileBranchId=3243807#l119)

**Pull request merged**